server/src/test/java/org/elasticsearch/common/util/concurrent/TaskExecutionTimeTrackingEsThreadPoolExecutorTests.java

```diff
@@ -93,14 +93,81 @@ public void testExecutionEWMACalculation() throws Exception {
         executor.awaitTermination(10, TimeUnit.SECONDS);
     }
 
-    public void testMaxQueueLatency() throws Exception {
+    /**
+     * Verifies that we can peek at the task in front of the task queue to fetch the duration that the oldest task has been queued.
+     * Tests {@link TaskExecutionTimeTrackingEsThreadPoolExecutor#peekMaxQueueLatencyInQueue}.
+     */
+    public void testFrontOfQueueLatency() throws Exception {
         ThreadContext context = new ThreadContext(Settings.EMPTY);
-        RecordingMeterRegistry meterRegistry = new RecordingMeterRegistry();
-        final var threadPoolName = randomIdentifier();
```

Comment:
Both of these were unused, so I took the opportunity to delete them. Does mess up the diff, a bit, though.